crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -863,7 +876,90 @@ where
             Some(recpts) => recpts,
             None => return Err(EthApiError::UnknownBlockNumber),
         };
-        build_transaction_receipt_with_block_receipts(tx, meta, receipt, &all_receipts)
+
+        #[cfg(feature = "optimism")]
+        let optimism_block_meta = {
+            let block = self
+                .provider()
+                .block_by_number(meta.block_number)?
+                .ok_or(EthApiError::UnknownBlockNumber)?;
+
+            let block_timestamp = block.timestamp;
+
+            let l1_block_info: Option<Arc<L1BlockInfo>> = reth_revm::optimism::parse_l1_info_tx(
+                &block.body.first().ok_or(EthApiError::InternalEthError)?.input()[4..],
+            )
+            .map_err(|_| EthApiError::InternalEthError)
+            .ok()
+            .map(Arc::new);
```

Comment:
why the Arc wrapper?

Reply:
Not sure - this was added in https://github.com/anton-rs/op-reth/pull/150, comments lead me to believe we thought it'd be shared among threads. Nice to have a reference count to be able to share among receipts in the `block_receipts` fn, though.